src/include/migraphx/check_shapes.hpp

```diff
@@ -235,11 +235,14 @@ struct check_shapes
     }
 
     /*!
-     * Check all shapes have the same layout.
+     * Check all shapes have the same layout, with minor differences allowed.
      */
-    const check_shapes& same_layout() const
+    const check_shapes& compatible_layout() const
```

Comment:
Let not make the changes here. This might cause problems with MIOpen. Instead we should just update auto_contiguous to insert contiguous even if its standard and not transposed instead:

```cpp
        if(not s.dynamic() and not s.standard() and not s.transposed() and s.elements() != 0)
        {
            auto c = m.insert_instruction(std::next(ins), make_op("contiguous"), ins);
            m.replace_instruction(ins, c);
        }
```

Which comes from [here](https://github.com/ROCm/AMDMIGraphX/blob/49fd94aff22084f8848886a7d058a43670e8afbc/src/auto_contiguous.cpp#L67).

Reply:
After some discussion, I reverted as @pfultz2 requested but with a correction to the Line 67 logic above.  Paul, please note that if this check is done incorrectly and a `contiguous`  gets added at the wrong time, it can result in an infinite loop.  You can see an example of this in commit `97a0b7e5b9a1bdcf223d680febd1dc99be9d6d1b` which also has a descriptive comment at this same point.